src/App.tsx

```diff
@@ -1,204 +1,128 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { todosService, USER_ID } from './api/todos';
+
+import { useEffect, useRef, useState } from 'react';
+
+import './styles/todoapp.scss';
+
+import { Todo } from './types/Todo';
+import { ErrorMessages } from './types/errorMessage';
+
+import { Header } from './components/Header';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoList } from './components/Main/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { FilterType } from './types/FilterType';
+
+function getFilteredTodosFromApi(todos: Todo[], filter: FilterType) {
+  const todoList = [...todos];
+
+  switch (filter) {
+    case FilterType.All:
+      return todoList;
+    case FilterType.Active:
+      return todoList.filter(todo => todo.completed === false);
+    case FilterType.Completed:
+      return todoList.filter(todo => todo.completed === true);
+    default:
+      throw new Error();
+  }
+}
+
+export const App = () => {
+  const todoFromServer = useRef<Todo[]>([]);
+  const [todoList, setTodoList] = useState<Todo[]>([]);
```

Comment:
```suggestion
  const [todoList, setTodoList] = useState<Todo[]>([]);
```
I don't understand why you need useRef
it's over engeneering - KISS
let's get rid of useRef